tmt/steps/report/polarion.py

```diff
@@ -213,93 +227,121 @@ def go(self, *, logger: Optional[tmt.log.Logger] = None) -> None:
+                "'--project-id' argument or by setting 'TMT_PLUGIN_REPORT_POLARION_PROJECT_ID' "
+                "environment variable.")
+
         # TODO: try use self.data instead - but these fields are not optional, they do have
         # default values, do envvars even have any effect at all??
         upload = self.get('upload', os.getenv('TMT_PLUGIN_REPORT_POLARION_UPLOAD'))
         use_facts = self.get('use-facts', os.getenv('TMT_PLUGIN_REPORT_POLARION_USE_FACTS'))
-        other_testrun_fields = [
-            'description', 'planned_in', 'assignee', 'pool_team', 'arch', 'platform', 'build',
-            'sample_image', 'logs', 'compose_id', 'fips']
-
-        xml_data = make_junit_xml(
-            phase=self,
-
-            # TODO: Explicitly use 'default' flavor until the 'polarion' flavor
-            # gets implemented in junit report plugin.
-            # flavor='polarion',
-            flavor='default',
 
-            include_output_log=self.data.include_output_log)
+        other_testrun_fields = [
+            'arch',
+            'assignee',
+            'build',
+            'compose_id',
+            'description',
+            'fips',
+            'logs',
+            'planned_in',
+            'platform',
+            'pool_team',
+            'sample_image']
+
+        testsuites_properties: dict[str, Optional[str]] = {}
 
-        # S314: Any potential xml parser vulnerability mitigation would require defusedxml package
-        xml_tree = ET.fromstring(xml_data)  # noqa: S314
-        properties = {
-            'polarion-project-id': project_id,
-            'polarion-user-id': PolarionWorkItem._session.user_id,
-            'polarion-testrun-title': title,
-            'polarion-project-span-ids': project_id}
         for tr_field in other_testrun_fields:
             param = self.get(tr_field, os.getenv(f'TMT_PLUGIN_REPORT_POLARION_{tr_field.upper()}'))
             # TODO: remove the os.getenv when envvars in click work with steps in plans as well
             # as with steps on cmdline
             if param:
-                properties[f"polarion-custom-{tr_field.replace('_', '')}"] = param
+                testsuites_properties[f"polarion-custom-{tr_field.replace('_', '')}"] = param
+
         if use_facts:
-            properties['polarion-custom-hostname'] = \
-                self.step.plan.provision.guests()[0].primary_address
-            properties['polarion-custom-arch'] = self.step.plan.provision.guests()[0].facts.arch
+            guests = self.step.plan.provision.guests()
+            try:
+                testsuites_properties['polarion-custom-hostname'] = guests[0].primary_address
+                testsuites_properties['polarion-custom-arch'] = guests[0].facts.arch
+            except IndexError as error:
+                raise tmt.utils.ReportError(
+                    'Failed to get the Polarion facts from '
+                    '`step.plan.provision.guests()`') from error
```

Comment:
To me as a user, `step.plan.provision.guests()` is nothing but a dark magic I don't understand. Did I do something wrong? Is Polarion broken? I bet we could provide some helpful tips for user reading the error messages :)

Reply:
Sure. Is it better now?